pkg/metainfo/kvmetainfo/stream.go

```diff
@@ -37,7 +37,7 @@ func (stream *readonlyStream) SegmentsAt(ctx context.Context, byteOffset int64,
 	return stream.Segments(ctx, index, limit)
 }
 
-func (stream *readonlyStream) segment(ctx context.Context, index int64) (segment storj.Segment, err error) {
+func (stream *readonlyStream) Segment(ctx context.Context, index int64) (segment storj.Segment, count int64, err error) {
```

Comment:
I'm not very familiar with kvmetainfo but I don't think its good idea to mix info about segment with retrieving storage node info with `StorageNodeInfo`. Maybe a separate method that will use `segment()` and later iterate over `segment.Pieces` (which contains NodeID).